src/rmem.c

```diff
@@ -0,0 +1,470 @@
+#include "rmem.h"
+
+
+typedef uintptr_t ptrcmp;
+
+static size_t _AlignSize(const size_t size, const size_t align)
+{
+    return (size + (align-1)) & -align;
+}
+
+
+/************* Memory Pool *************/
+
+static void _RemoveNode(struct MemNode **const node)
+{
+    ((*node)->prev != NULL)? ((*node)->prev->next = (*node)->next) : (*node = (*node)->next);
+    ((*node)->next != NULL)? ((*node)->next->prev = (*node)->prev) : (*node = (*node)->prev);
+}
+
+struct MemPool MemPool_Create(const size_t size)
+{
+    struct MemPool mempool = {0};
+    if( size==0UL )
+        return mempool;
+    else {
+        // align the mempool size to at least the size of an alloc node.
+        mempool.stack.size = _AlignSize(size, sizeof(struct MemNode));
+        mempool.stack.mem = malloc(1 + mempool.stack.size*sizeof *mempool.stack.mem);
+        if( mempool.stack.mem==NULL ) {
+            mempool.stack.size = 0UL;
+            return mempool;
```

Comment:
factor out the `return mempool` instead of repeating it.

Reply:
again, this is for branch prediction optimization.
I discovered this when I was working on my virtual machine project named Tagha.

Reply:
That might have been the case in your other project. I fail to see how this would affect branch prediction _here_ anyhow. If you absolutely want to mark branches, use a macro around `__builtin_expect` (check `<linux/kernel.h>` `likely` and `unlikely` macros).